File: golang/quickSort.go
package main

import "fmt"

func quickSort(array []int) []int {

	if len(array) < 2 {
		return array
	}

	pivo := array[0]
	var menores []int
	var maiores []int

	for _, i := range array[1:] {
		if i <= pivo {
			menores = append(menores, i)
		}
	}

	for _, i := range array[1:] {
		if i > pivo {
			maiores = append(maiores, i)
		}
	}

	return append(append(quickSort(menores), pivo), quickSort(maiores)...)

}

func main() {

	x := quickSort([]int{10, 5, 3, 2, 8, 7, 4})

	fmt.Println(x)

}

File: python/quickSort.py
def quicksort(array: list) -> list:

    if len(array) < 2:
        return array
    
    
    pivo = array[0]

    menores = [i for i in array[1:] if i <= pivo]
    maiores = [i for i in array[1:] if i > pivo]

    return quicksort(menores) + [pivo] + quicksort(maiores)

print(quicksort([10, 5, 2, 3, 8, 4, 9]))